lib/mp4/transmuxer.js

```diff
@@ -20,7 +20,48 @@ var H264Stream = require('../codecs/h264').H264Stream;
 var VideoSegmentStream, AudioSegmentStream, Transmuxer, CoalesceStream;
 
 // Helper functions
-var collectDtsInfo, clearDtsInfo, calculateTrackBaseMediaDecodeTime;
+var collectDtsInfo, clearDtsInfo, calculateTrackBaseMediaDecodeTime, compareArrays, sumByteLengths;
+
+/**
+ * Compare two arrays (event typed) for same-ness
+ */
+compareArrays = function(a, b) {
+  var
+    i,
+    currentObj,
+    sum = 0;
+
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  // sum the byteLength's all each nal unit in the frame
+  for (i = 0; i < a.length; i++) {
+    if (a[i] !== b[i]) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+/**
+ * Sum the `byteLength` of a specific property in an array of objects
```

Comment:
This could also be done as a map() then a sum. If you like sticking with this method, could you elaborate a bit more on this function level comment?